sdk/python/src/sangria_sdk/client.py

```
@@ -116,5 +128,96 @@ async def _settle_payment(
+            network=result.get("network"),
+            payer=result.get("payer"),
+            error_reason=result.get("error_reason"),
+            error_message=result.get("error_message"),
+        )
+
+    def create_settle_fn(
+        self,
+        max_price: float,
+    ) -> tuple[Callable[[float, Any], Settled], Callable[[], tuple[float, Any] | None]]:
+        if not isinstance(max_price, (int, float)) or not math.isfinite(max_price) or max_price <= 0:
+            raise ValueError("Sangria: max_price must be a positive finite number")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)